packages/app/src/client/views/login/index.tsx

```diff
@@ -175,17 +259,44 @@ const Login = (props: { state?: State }) => {
   );
 };
 
-const LoginPage = () => {
+function LoginPageParent() {
+  // Uninitialized state will cause Child to error out
+  const [loginConfig, setLoginConfig] = useState<LoginConfigInterface | undefined>();
+  useEffect(() => {
+    (async () => {
+      // Without this early exit criterion, `fetchLoginConfig()` is going to
+      // be called with every render cycle.
```

Comment:
@jgehrcke To avoid this, `useEffect` has the notion of dependencies, and will only be run when a dependency changes.

The react documentation on this is quite thorough and well worth a read: https://reactjs.org/docs/hooks-effect.html

Most relevant is "skipping effects" -> https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects

In the case here, changing the `useEffect` hook to:
```
useEffect(() => {
....
}, [loginConfig])
```
(which React desires -> all variables used inside the hook should be added to the dependency array so the hook only is rerun when a dependency's value changes). This hook will now only be called on the first render when `loginConfig === undefined` and then it will be called once more when `loginConfig === <LoginConfigInterface>`. It will also get called once more when the component is unmounted so any cleanup can happen https://reactjs.org/docs/hooks-effect.html#effects-with-cleanup

Reply:
lovely, thanks!